azure-quantum/azure/quantum/projectq/backends/ionq.py

```diff
@@ -29,35 +33,54 @@
 
 
 class IonQBackend(_IonQBackend):
-    projectq_backend_name = None
-    azure_quantum_backend_name = None
+    projectq_backend_name: str = None
+    azure_quantum_backend_name: str = None
 
     def __init__(
         self, 
-        use_hardware=False, 
-        num_runs=100, 
-        verbose=False, 
-        device="ionq_simulator", 
-        retrieve_execution=None
+        use_hardware: bool=False,
+        num_runs: int=100,
+        verbose: bool=False,
+        device: str="ionq_simulator",
+        num_retries: int = 3000,
+        interval: int = 1,
+        retrieve_execution: str=None
     ):
-        """Base class for interfacing with an IonQ backend in Azure Quantum"""
+        """Base class for interfacing with an IonQ backend in Azure Quantum
+
+        :param use_hardware: Whether or not to use real IonQ hardware or just a simulator. If False, the
+            ionq_simulator is used regardless of the value of ``device``. Defaults to False.
+        :param num_runs: Number of times to run circuits. Defaults to 100.
+        verbose: If True, print statistics after job results have been collected. Defaults to
+            False.
+        :param token: An IonQ API token. Defaults to None.
+        :param device: Device to run jobs on.  Supported devices are ``'ionq_qpu'`` or
+            ``'ionq_simulator'``.  Defaults to ``'ionq_simulator'``.
+        :param num_retries: Number of times to retry fetching job results after it has been submitted. Defaults
+            to 3000.
+        :param interval: Number of seconds to wait inbetween result fetch retries. Defaults to 1.
+        :param retrieve_execution: An IonQ API Job ID.  If provided, a job with this ID will be
+            fetched. Defaults to None.
+        """
         logger.info("Initializing IonQBackend for ProjectQ")
 
         super().__init__(
             use_hardware=use_hardware, 
             num_runs=num_runs, 
             verbose=verbose, 
-            device=device, 
+            device=device,
+            num_retries=num_retries,
```

Comment:
'num_retries` and `interval` are client side properties in ProjectQ which used in retry mechanism when job failures.

If we choose to include them in the azure-quantum backends, we need to implement retry mechanism in `submit_job()` method (simply by using `for-loop` and `sleep`). Let me know if it makes sense.

Reply:
I think we can indeed drop these from the engine.